src/app/modules/main/wallet_section/module.nim

```diff
@@ -132,29 +132,28 @@ method getCurrentCurrency*(self: Module): string =
 method setTotalCurrencyBalance*(self: Module) =
   var addresses: seq[string] = @[]
   let walletAccounts = self.controller.getWalletAccounts()
-  if self.filter.excludeWatchOnly:
-    addresses = walletAccounts.filter(a => a.walletType != "watch").map(a => a.address)
-  else:
+  if self.controller.isIncludeWatchOnlyAccount():
     addresses = walletAccounts.map(a => a.address)
+  else:
+    addresses = walletAccounts.filter(a => a.walletType != "watch").map(a => a.address)
 
   self.view.setTotalCurrencyBalance(self.controller.getCurrencyBalance(addresses))
 
 method notifyFilterChanged(self: Module) =
-  self.overviewModule.filterChanged(self.filter.addresses, self.filter.chainIds, self.filter.excludeWatchOnly, self.filter.allAddresses)
+  let includeWatchOnly = self.controller.isIncludeWatchOnlyAccount()
+  self.overviewModule.filterChanged(self.filter.addresses, self.filter.chainIds, includeWatchOnly, self.filter.allAddresses)
   self.assetsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
   self.collectiblesModule.filterChanged(self.filter.addresses, self.filter.chainIds)
   self.transactionsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
-  self.accountsModule.filterChanged(self.filter.addresses, self.filter.chainIds, self.filter.excludeWatchOnly)
+  self.accountsModule.filterChanged(self.filter.addresses, self.filter.chainIds)
   self.sendModule.filterChanged(self.filter.addresses, self.filter.chainIds)
-  self.view.filterChanged(self.filter.addresses[0], self.filter.excludeWatchOnly, self.filter.allAddresses)
+  self.view.filterChanged(self.filter.addresses[0], includeWatchOnly, self.filter.allAddresses)
```

Comment:
I am not sure it `self.filter.addresses` can be an empty seq or not, but accessing directly by index without check is always dangerous.